src/storage/aggregator.cc

```diff
@@ -786,82 +821,128 @@ bool MinMaxBaseAggregator::DecodeAggrVal(const int8_t* row_ptr, AggrBuffer* buff
     return true;
 }
 
+bool MinMaxBaseAggregator::UpdateAggrVal(const codec::RowView& row_view, const int8_t* row_ptr, AggrBuffer* aggr_buffer,
+                   const std::string& key, bool reverse) {
+    return true;
+}
+
+bool MinMaxBaseAggregator::RebuildExtremumAggrBuffer(const codec::RowView& row_view,
+                                                     AggrBuffer* aggr_buffer,
+                                                     const std::string& key) {
+    if (base_table_ == nullptr) {
+        PDLOG(ERROR, "base table is nullptr, cannot update MinAggr table");
+        return false;
+    }
+    auto it = base_table_->NewTraverseIterator(this->GetIndexPos());
+    int64_t ts_begin = aggr_buffer->ts_begin_;
+    int64_t ts_end = aggr_buffer->ts_end_;
+    it->Seek(key, ts_end);
+    aggr_buffer->clear();
+    if (this->GetWindowType() == WindowType::kRowsRange) {
+        aggr_buffer->ts_begin_ = ts_begin;
+        aggr_buffer->ts_end_ = ts_end;
+    }
+    while (it->Valid() && it->GetKey() >= (uint64_t)ts_begin) {
+        if (it->GetKey() <= (uint64_t)ts_end) {
+            const int8_t* base_row_ptr = reinterpret_cast<const int8_t*>(it->GetValue().data());
+            if (!UpdateAggrVal(row_view, base_row_ptr, aggr_buffer, key, false)) {
+                PDLOG(ERROR, "Failed to update aggr Val during rebuilding Extermum aggr buffer");
+                return false;
+            }
+            if (this->GetWindowType() == WindowType::kRowsRange) {
+                if (aggr_buffer->ts_begin_ == -1) {
+                    // the first record.
+                    aggr_buffer->ts_end_ = it->GetKey();
+                }
+                aggr_buffer->ts_begin_ = it->GetKey();
+            }
```

Comment:
why check the `kRowsRange` again here? you intend to check `kRowsNum`?

as we keep the `ts_end_` and `ts_start_` as previous buffer, can remove the code